edx/edx-enterprise
# [ENT-616] Fix course modal UI issues (...with Bootstrap & SASS/SCSS)

**Description:** This PR primarily seeks to fix the course modal (and the DSC confirmation modal) that we have in our enterprise learner flow. In the process, it does the following:

1. Uses [edx-bootstrap](https://www.github.com/edx/edx-bootstrap) for our course and DSC confirmation modals; this prepares us to use the latest, most modern technology available and stay up-to-date with the direction edX is taking in front-end technologies, and also makes implementation and maintenance of our modals easier and much more guaranteed to be mobile responsive.
1. Rewrites all of our CSS in SASS/SCSS, helping us lessen code and consolidating our styles further. This is required if we want to take advantage of the features in `edx-bootstrap` (as it's fully in SASS/SCSS itself because Bootstrap 4 is, too).
1. Sets up a Webpack SASS/SCSS bundling pipeline, required for easily compiling our new SASS/SCSS into CSS for use; this also prepares us for when we become a service and need to bundle our static content optimally on provision.
1. Updates any necessary JS to account for the above changes.

**JIRA:** [ENT-616](https://openedx.atlassian.net/browse/ENT-616)

**Merge deadline:** Before end of sprint.

**Installation instructions:** If you want to develop on this, you **musn't** forget to `npm install` initially, and do `make static` when changing static content. Right now this matters, but in the future when we become a service, static asset generation should happen automatically when running the service in a container. But since we're currently being installed through `pip`, we have to do it manually.

**Testing instructions:**

Since this rewrites all of our CSS in SASS/SCSS, it would be wise to go through our landing pages, the DSC page, and the manage-learners admin portal to make sure everything looks and feels the same.

**NOTE**: Bootstrap was *only* used for the modal, so this does not necessarily make our overall pages any more mobile responsive; it just really applies to the modals.

For the **course enrollment landing page (1)**, **program enrollment landing page (2)**, **DSC confirmation page (3)**, and the **manage learners admin portal (4)**, ensure the following where they apply:
- Go to the relevant page.
- Make sure that styling looks the same as it did before this PR.
- Ensure that the page is as mobile responsive as it used to be.
- If a modal exists, open it, and see that it covers any issues enlisted on the ticket, and especially that it is mobile responsive.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
